File: pset6/credit/credit.py
from cs50 import get_int

# TO PROMPT USER INPUT
while True:
    card = input("Enter card number :")
    if(card.isnumeric()):
        break

checksum = 0
# CONVERTING card INTO LIST
card = list(card)
length = len(card)

# TO CHANGE ALL VALUES OF card INTO INT
for i in range(length):
    card[i] = int(card[i])

# CALCULATING CHECKSUM
for i in range(length - 2, -1, -2):
    scc = (card[i]) * 2

    if (scc > 9):
        checksum += scc % 10
        scc = scc // 10

    checksum += scc

for i in range(length - 1, -1, -2):
    checksum += (card[i])

# VALIDATING THE CARD
if checksum % 10 == 0:
    if length == 15 and card[0] == 3:
        if card[1] == 4 or card[1] == 7:
            print("AMEX")

    elif length == 16 and card[0] == 5:
        if card[1] == 1 or card[1] == 2 or card[1] == 3 or card[1] == 4 or card[1] == 5:
            print("MASTERCARD")

    elif (length == 13 or length == 16) and card[0] == 4:
        print("VISA")

    else:
        print("VALID\nCannot determine the company.")
else:
    print("INVALID")
# END OF PROGRAM

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    //Initialising the variables.
    long long num;
    int scc, checkSum = 0, i = 0, j = 0;
    int cc[20], cc_number[20];

    printf("Number: ");
    num = get_long_long();

    //to convert and store the given number in an array backwards.
    while (num != 0)
    {

        cc[i] = num % 10;
        num = num / 10;
        i++;
    }

    int length = i;
    j = i - 1;
    i = 0;

    /*To store the numbers of cc array in cc_number array from the back
    so that it's stored forwards in array format*/
    while (i < length && j >= 0)
    {
        cc_number[i] = cc[j];
        i++, j--;
    }

    /*Multiplying every other digit by 2, starting with the number’s second-to-last digit,
    and then adding those product's digits together.*/
    for (i = length - 2; i >= 0; i -= 2)
    {
        scc = cc_number[i] * 2;

        if (scc > 9)
        {
            checkSum += scc % 10;
            scc = scc / 10;
        }
        checkSum = checkSum + scc;
    }

    //Adding the remaining digits which weren't multiplied by 2 to the previous sum.
    for (i = length - 1; i >= 0; i -= 2)
    {
        checkSum += cc_number[i];
    }

    //Checking if its VALID or INVALID.
    if (checkSum % 10 == 0)
    {
        //Checking if its an AMEX card.
        if (length == 15 && cc_number[0] == 3)
        {
            if (cc_number[1] == 4 || cc_number[1] == 7)
            {
                printf("AMEX\n");
            }

        }
        //Checking if its a MASTERCARD.
        else if (length == 16 && cc_number[0] == 5)
        {
            if (cc_number[1] == 1 || cc_number[1] == 2 || cc_number[1] == 3 || cc_number[1] == 4
                || cc_number[1] == 5)
            {
                printf("MASTERCARD\n");
            }
        }
        //Checking if its a VISA card.
        else if ((length == 13 || length == 16) && cc_number[0] == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("VALID\nCannot determine the company.\n");
        }


    }
    else
    {
        printf("INVALID\n");
    }
    return 0;
}

//END OF PROGRAM.